crates/volta-core/src/command.rs

```diff
@@ -28,3 +31,52 @@ cfg_if! {
         }
     }
 }
+
+/// Rebuild command against given PATH
+///
+/// On Windows, we need to explicitly use an absolute path to the executable,
+/// otherwise the executable will not be located properly, even if we've set the PATH.
+/// see: https://github.com/rust-lang/rust/issues/37519
+///
+/// This function will try to find the executable in the given path and rebuild
+/// the command with the absolute path to the executable.
+pub fn rebuild_command<S: AsRef<OsStr>>(command: Command, path: S) -> Fallible<Command> {
+    debug!("PATH: {}", path.as_ref().to_string_lossy());
+
+    #[cfg(not(windows))]
+    {
+        let mut command = command;
+        command.env("PATH", path.as_ref());
+        Ok(command)
+    }
+
+    #[cfg(windows)]
+    {
+        let args = command.get_args().collect::<Vec<_>>();
+        //          cmd /c <name> [...other]
+        // args_idx     0  1      2..
+        let name = args.get(1).expect("should have the name");
+
+        if let Ok(mut paths) = which::which_in_global(name, Some(&path)) {
+            if let Some(exe) = paths.next() {
+                let mut new_command = create_command(exe);
+                let envs = command
```

Comment:
1. Why does this need to filter the environment variables like this, rather than maintaining the same environment variables as the previous version of the command?
2. Can you add a comment explaining that here? That will help us remember (and other contributors understand)!

Reply:
Agreed on adding a comment!

I believe this needs to filter because the output of [`Command::get_envs`](https://doc.rust-lang.org/std/process/struct.Command.html#method.get_envs) includes `None` values for variables that were explicitly removed. I don't think we do any explicit removing of environment variables currently, but possibly what we want to do is mirror the same behavior:

```rust
for (key, maybe_value) in command.get_envs() {
    match maybe_value {
        Some(value) => command.env(key, value),
        None => command.env_remove(key),
    }
}
```